Added documentation regarding the groupby-row selectors (#2255)
IgniteUI/igniteui-docfx: en/components/grid/groupby.md

# Grid Group By

> A Group by behavior in an Angular Material table or UI grid creates grouped data rows based on the column values. The Group By in [`igxGrid`]({environment:angularApiUrl}/classes/igxgridcomponent.html) allows for visualizing the groups in a hierarchical structure. The grouped data rows can be expanded or collapsed and the order of grouping may be changed through the UI or API. When row selection is enabled groupby row selector would be rendered on the most left part of each groupby row. In case the **rowSelection** property is set to single the checkboxes would be disabled and would only serve as an indicator of the group into which the selection is placed. If the **rowSelection** property is set to multiple clicking over the groupby row selectors would select all records belonging to this group.

Reviewer: ```suggestion
A Group By behavior in an Angular Material table or UI grid creates grouped data rows based on the column values. The Group By in [`igxGrid`]({environment:angularApiUrl}/classes/igxgridcomponent.html) allows for visualizing the groups in a hierarchical structure. The grouped data rows can be expanded or collapsed and the order of grouping may be changed through the UI or API. When Row Selection is enabled, a Group By row selector is rendered in the left-most area of the group row. In case the **rowSelection** property is set to single, checkboxes are disabled and only serve as an indication for the group where selection is placed. If the **rowSelection** property is set to multiple, clicking over the Group By row selector selects all records belonging to this group.
```

Groups can be created expanded (***default***) or collapsed and the expansion states would generally only contain the state opposite to the default behavior. You can control whether groups should be created expanded or not through the [`groupsExpanded`]({environment:angularApiUrl}/classes/igxgridcomponent.html#groupsexpanded) property.

### Select/Deselect all rows in a group API

> Selecting/Deselecting all rows in a certain group is available through the [`selectAllRowsInGroup`]({environment:angularApiUrl}/classes/igxgridcomponent.html#selectallrowsingroup) and [`deselectAllRowsInGroup`]({environment:angularApiUrl}/classes/igxgridcomponent.html#deselectallrowsingroup) API methods.

Reviewer: ```suggestion
Selecting/Deselecting all rows in a group is available through the [`selectAllRowsInGroup`]({environment:angularApiUrl}/classes/igxgridcomponent.html#selectallrowsingroup) and [`deselectAllRowsInGroup`]({environment:angularApiUrl}/classes/igxgridcomponent.html#deselectallrowsingroup) API methods.
```

> The below code example can be used to select all rows within a group by the group record instance using [`selectAllRowsInGroup`]({environment:angularApiUrl}/classes/igxgridcomponent.html#selectallrowsingroup) method. Additionally, as a second parameter of this method, you can pass a boolean property which corresponds to whether the previous row selection will be cleared or not. By default the previous selection will be preserved.

Reviewer: ```suggestion
The code snippet below can be used to select all rows within a group using the group record instance [`selectAllRowsInGroup`]({environment:angularApiUrl}/classes/igxgridcomponent.html#selectallrowsingroup) method. Additionally, the second parameter of this method is a boolean property through which you may choose whether the previous row selection will be cleared or not. The previous selection is preserved by default.
```

</ng-template>
```

### Group Row Selector Templates

> As mentioned above the group row without its expand/collapse UI is fully templatable. To create a custom groupby row selector template, within the Grid, declare an `<ng-template>` with `igxGroupByRowSelector` directive. From the template you can access the implicitly provided context variable, with properties that give you information about the groupby-row's state.

Reviewer: ```suggestion
As mentioned above the group row except for the expand/collapse UI is fully templatable. To create a custom Group By row selector template within the Grid, declare an `<ng-template>` with `igxGroupByRowSelector` directive. From the template, you can access the implicitly provided context variable, with properties that give you information about the Group By row's state.
```

The `selectedCount` property shows how many of the group records are currently selected while `totalCount` shows how many records belong to the group.

```html
<ng-template igxGroupByRowSelector let-groupByRowContext>
    {{ groupByRowContext.selectedCount }} / {{ groupByRowContext.totalCount  }}
</ng-template>
```

> The `groupRow` property property returns a reference to the group row.

Reviewer: ```suggestion
The `groupRow` property returns a reference to the group row.
```

```html
<ng-template igxGroupByRowSelector let-groupByRowContext>
    <div (click)="handleGroupByRowSelectorClick($event, groupByRowContext.groupRow)">Handle groupRow</div>
</ng-template>
```

> The `selectedCount` and `totalCount` properties can be used to determine if the groupby row selector should be checked or partially checked (indeterminate).

Reviewer: ```suggestion
The `selectedCount` and `totalCount` properties can be used to determine if the Group By row selector should be checked or indeterminate (partially selected).
```
